In [1]:
import torch
import torch.nn as nn
import torch.optim as optim
import torch.nn.functional as F
from torch.autograd import Variable
import torch.backends.cudnn as cudnn
# cudnn.benchmark = True

import torchvision.transforms as transforms
import torchvision.datasets as datasets

import problems as pblm
from trainer import *
import math
import numpy as np

In [2]:
batch_size = 20
test_log = open("test.log", "w")

In [3]:
train_loader, test_loader = pblm.mnist_loaders(batch_size)

Let's initialize the model and load weights from the robust training

In [5]:
model = pblm.mnist_model().cuda()

In [6]:
model.load_state_dict(torch.load('mnist.pth'))

# Evaluation

In [11]:
epsilon = 0.1
evaluate_robust(test_loader, model, epsilon, 1, test_log, verbose=0)

C:\Program Files\Anaconda3\lib\site-packages\torch\nn\functional.py:52: UserWarning: size_average and reduce args will be deprecated, please use reduction='elementwise_mean' instead.
  warnings.warn(warning.format(ret))



 * Robust error 0.058	Error 0.018


0.0582

For the MNIST test set, only 5.8% of examples can be fooled when $\epsilon$=0.1. The model has the standard error of 1.8%, that is 98.2% examples are classified correctly. These numbers match the number reported in the paper (Table 1, page 8)